src/components/file-upload-header/file-upload-header.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import ButtonUnstyled from "@mui/base/ButtonUnstyled";
+import { Box, styled } from "@mui/material";
+import { InputTitle } from "..";
+
+const StyledFileUploadHeader = styled(Box, {
+  name: "WmeFileUploadHeader",
+  slot: "Root",
+})({
+  alignItems: "center",
+  display: "flex",
+  justifyContent: "space-between",
+  "& .WmeErrorText": {
+    cursor: "pointer",
+    marginLeft: "auto",
+  },
+});
+
+const StyledButton = styled(ButtonUnstyled, {
+  name: "WmeFileUploadHeaderButton",
+  slot: "Root",
+})(({ theme }) => ({
+  background: "none",
+  border: "none",
+  color: theme.palette.error.main,
+  cursor: "pointer",
+  margin: 0,
+  padding: 0,
+}));
+
+export interface FileUploadHeaderProps {
+  buttonText?: string;
+  fileUploaded?: boolean;
+  labelText?: string;
+  onRemove?: React.MouseEventHandler<HTMLButtonElement>;
+}
+
+const FileUploadHeader: React.FC<FileUploadHeaderProps> = ({
```

Comment:
Does this need a corresponding story?

Reply:
I could go either way honestly since it's only used in `FileUpload` I skipped it, but I also am big on documentation

Reply:
let's leave it as is for now, we can add it later if we feel it's needed

Reply:
Didn't feel right to NOT have documentation, so I've just added 😄